src/tsolvers/lasolver/LASolver.cc

```diff
@@ -858,4 +858,79 @@ PTRef LASolver::getIntegerInterpolant(std::map<PTRef, icolor_t> const& labels) {
 void LASolver::printStatistics(std::ostream & out) {
     TSolver::printStatistics(out);
     laSolverStats.printStatistics(out);
+}
+
+void LASolver::collectEqualitiesFor(const vec<PTRef> & vars, vec<PTRef> & equalities, std::unordered_set<PTRef, PTRefHash> const & knownEqualities) {
+    struct DeltaHash {
+        std::size_t operator()(Delta const & d) const {
+            FastRationalHash hasher;
+            return (hasher(d.R()) ^ hasher(d.D()));
+        }
+    };
+
+    std::unordered_map<Delta, vec<PTRef>, DeltaHash> eqClasses;
+    for (PTRef var : vars) {
+        if (logic.isNumConst(var)) {
+            eqClasses[logic.getNumConst(var)].push(var);
+        } else {
+            assert(logic.isNumVar(var));
+            if (not laVarMapper.hasVar(var)) { // LASolver does not have any constraints on this LA var
+                continue;
+            }
+            LVRef v = laVarMapper.getVarByPTId(logic.getPterm(var).getId());
+            auto value = simplex.getValuation(v);
+            eqClasses[value].push(var);
+        }
+    }
+
+    for (auto const & entry : eqClasses) {
+        auto const & equivalentVars = entry.second;
+        for (int i = 0; i < equivalentVars.size(); ++i) {
+            for (int j = i + 1; j < equivalentVars.size(); ++j) {
+                PTRef eq = logic.mkEq(equivalentVars[i], equivalentVars[j]);
+                if (knownEqualities.count(eq) == 0) {
+                    equalities.push(eq);
+                }
+            }
+        }
+    }
+
+    // Hack to ensure that when model is computed, delta is not picked so that it merges some of the equivalence
```

Comment:
To improve this, we could lower the maximal value of Delta in Simplex from `1` to e.g. `1/2`.

Another possibility that came to my mind, would be to compute the real values first in the current model, and use the real values to find equalities. Not sure if this would be faster or slower...